Sources/XCMetricsClient/XCMetrics.swift

```diff
@@ -109,6 +110,14 @@ public struct XCMetrics: ParsableCommand {
     @Option(name: [.customLong("skipNotes")], help: "Notes found in logs won't be processed")
     public var skipNotes: Bool = false
 
+    /// An optional authorization/token header **key** to be included in the upload request. Must be used in conjunction with `authorizationValue.`
+    @Option(name: [.customLong("authorizationKey"), .customShort("k")], help: "An optional authorization header key to be included in the upload request e.g 'Authorization' or 'x-api-key' etc. Must be used in conjunction with `authorizationValue`")
+    public var authorizationKey: String?
+
+    /// An optional authorization/token header **value** to be included in the upload request. Must be used in conjunction with `authorizationKey.`
+    @Option(name: [.customLong("authorizationValue"), .customShort("a")], help: "An optional authorization header value to be included in the upload request e.g 'Basic YWxhZGRpbjpvcGVuc2VzYW1l' or `hYDqG78OIUDIWKLdwjdwhdu8` etc. Must be used in conjunction with `authorizationKey`")
+    public var authorizationValue: String?
```

Comment:
I'm wondering why we need two specific arguments for a key and a value? Since everything else is very generic already, why not making this `additionalHeaders`, accept a dictionary encoded as a String (i.e: `--additionalHeaders "Key1:Value1,Key2:Value2"` or similar) and allow all kinds of headers to be injected. In this way we could also support multiple headers, since from what I can read now, this supports only one pair.

Reply:
So the main issue with this is that `:` and `,` can appear within the values of the headers so it would make it very difficult to differentiate between what was part of a value and what was a separator character e.g imagine a password like `9#Gshg&,GFNda@B@`, a comma exists which would cause the key-value split to break.

We could possibly force the user to use the hex value of the character within the value field, but I would need to double check that. Unless you have any additional thoughts?